src/System.Net.WebSockets.Client/src/System/Net/WebSockets/WebSocketHandle.Managed.cs

```diff
@@ -223,10 +223,12 @@ public async Task ConnectAsyncCore(Uri uri, CancellationToken cancellationToken,
                 Abort();
                 response?.Dispose();
 
-                if (exc is WebSocketException)
+                if (exc is WebSocketException ||
+                    (exc is OperationCanceledException && cancellationToken.IsCancellationRequested))
```

Comment:
Is there any risk of getting an `OperationCanceledException` for a different `CancellationToken`?

Reply:
> Is there any risk of getting an OperationCanceledException for a different CancellationToken?

It's certainly possible, e.g. for an Abort() call, which is why we check whether this cancellation token had cancellation requested.  If something else caused cancellation, then from the perspective of this call it wasn't the one to cancel it, and so it continues to wrap the exception.  If something else caused cancellation but this was also canceled, then it doesn't matter which actually triggered things to cancel, and so we allow it through.